### DISCUSSION TITLE
Can someone provide list of attacks supported for Object detection?

### DISCUSSION THREAD

**Discussion by phanisai22** [Trusted-AI/adversarial-robustness-toolbox#1573] [Q&A]
Hello,

So far I was able to run PGD, FGSM, basic Iterative method attacks on FRCNN pytorch Model.

Would like to know if there are any other supported attacks for this FRCNN pytorch Model. Specifically is there a way to implement Deepfool and Newtonfool.

Thanks.

**Comment by beat-buesser** [comment]
Hi @phanisai22 I agree, a better way to filter or list attacks available for each ML task would be great.

Currently ART has `RobustDPatch` as a patch attack against object detection and with ART 1.10 it will soon provide support for object detection in `AdversarialPatchPyTorch`.

You also should be able to apply the black-box attacks if you transform the object detection problem into a binary classification problem.

About your question on Deepfool and Newtonfool, I think the main question is how to extend the actual algorithm, not just the implementation, of Deepfool for object detection and make it work with object detection labels. What would be your main attack goal?

**Comment by phanisai22** [comment]
>You also should be able to apply the black-box attacks if you transform the object detection problem into a binary classification problem 

How can we do this transformation? any examples?

And Yes, I think you are right about extending the algorithm and not just the implementation, But It seems, there are not many attacks available for OD (unlike for ImageNet classification models). Any other set of attacks you would recommend for OD (if DeepFool, NewtonFool, etc are not possible)? I tried PGD, FGSM, RobustDPatch, Basic Iterative method and will look into AdversarialPatchPyTorch.

The goal is just to perform as many attacks and defenses as possible for a OD model XD. Targeted, untargeted, whitebox, blackbox anything. We are studying on what all attacks and defenses available for the case of OD.

**Reply by beat-buesser** [reply]
@phanisai22 That sounds interesting, let us know when you are publishing the results.

> How can we do this transformation? any examples?

To apply black-box attacks you could follow a similar approach as in this ART notebook on Tesseract OCR: https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/notebooks/classifier_blackbox_tesseract.ipynb

**Reply by phanisai22** [reply]
I have one more question, Is there a support for yolo models? I would like to run the available attacks on a yolo model.

**Reply by beat-buesser** [reply]
We don't have estimators for Yolo models in ART 1.10, but we are planning to add support in ART 1.11.
